grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy

```diff
@@ -203,4 +203,76 @@ class PrepaymentInvoiceService {
         Integer quantityInversed = orderItem.allInvoiceItems.findAll { it.inverseItem }.sum { it.quantity } ?: 0
         return prepaymentItem.quantity > quantityInversed ? prepaymentItem.quantity - quantityInversed : 0
     }
+
+    /**
+     * Updates invoice item's quantity and amount and finds related inverse item and updates quantity and amount
+     * */
+    void updateInvoiceItemQuantity(String itemId, Integer quantity) {
+        InvoiceItem invoiceItem = InvoiceItem.get(itemId)
+        if (quantity > invoiceItem.quantity) {
```

Comment:
Actually, this should be checking the quantity available to invoice from the shipment item instead of the current invoice item's quantity. Will fix that on Monday.

Reply:
Also, if you're checking shipment items quantity be weary of the case where an item is split across multiple shipments